### ISSUE TITLE
Preview GitHub activity in notifications without marking them as read

### ISSUE THREAD

**Issue by thejameskyle** [sindresorhus/refined-github#273]
Using the [GitHub Notifications API](https://developer.github.com/v3/activity/notifications/) it should be possible to create a little popover like the one below to show unseen activity using the `last_read_at` date. 

![screen shot 2016-07-26 at 1 09 13 pm](https://cloud.githubusercontent.com/assets/952783/17153785/103e2104-5333-11e6-824a-73b5eb18abda.png)

One of the biggest problems I have with lots of GitHub notifications is viewing them without marking them as read. Emails help this but they tend to get buried and are not organized by `user/repo`. GitHub notifications could use some love

**Comment by DrewML** [comment]
Oooh, that's a pretty good idea. I like it! This would require adding a settings panel, though (so users can enter their GitHub API key for private repo access). Sindre has tried to avoid adding a settings panel.

@sindresorhus thoughts? We _could_ also add this for only public repos, but that might be a bit of a weird experience from a UX perspective when some items work on hover and others don't.

**Comment by thejameskyle** [comment]
Public repos would be enough for me to be happy, those are the repos that have an unmanageable amount of notifications

**Comment by paulmolluzzo** [comment]
This is a really neat idea! If we go down the path of only showing this for public repos it might be worth indicating the ones that you can hover to preview. Maybe include the `eye` icon for the target to `:hover` on?

**Comment by sindresorhus** [comment]
Love it! I have the same issue as @thejameskyle. We can do public repos first and see.

**Comment by DrewML** [comment]
Started working on this tonight. I should have spotted this sooner, but the "Notifications API" requires an authentication token (which makes sense).

On a positive note, the link to the issue or PR in the DOM does include the comment ID that triggered the notification, so we can fetch that page >> grab the comment + all after >> display in the UI. This means we can support private repos, too, since we'll be piggybacking on the users current session.

**Comment by sindresorhus** [comment]
Reply

**Comment by DrewML** [comment]
Damn - hitting the URI at all dismisses the notification. I think the only way forward with this feature is providing users a way to add a GH API key.

**Comment by DrewML** [comment]
What do you think @sindresorhus? If we add the ability to enter an API key, it might open up an avenue to some other cool features we wouldn't be able to pull off without the API.

**Comment by sindresorhus** [comment]
Sure

**Comment by jamestalmage** [comment]
Ooh. If that's on the table, let's add a marker on the notifications page that differentiates notifications where you are `@`-mentioned. I would love to be able to prioritize what I  read that way

**Comment by DrewML** [comment]
@jamestalmage Want to open a separate issue for that one? I can do the API-key work for this PR first, get that committed, and then we'll be able to work on this issue + your idea in parallel.

**Comment by DrewML** [comment]
Still have plans to finish out what I started on this. Just hitting some timing issues w/ moving + priorities on other projects.

**Comment by vdemedes** [comment]
Hey guys, I've implemented a different approach to this problem, the way I always wanted it to be - check out PR #284 and let me know what you think over there!

Idea by @thejameskyle is great and certainly looks nice, but sometimes I want to give a thread another glance to understand the context of another person's reply. Because of that, a popover wouldn't work for me. Instead, I decided to add "Mark as unread" buttons to the sidebar on every issue/pr page:

<img width="321" alt="screen shot 2016-08-20 at 2 46 08 pm" src="https://cloud.githubusercontent.com/assets/697676/17831204/437d2aa8-66eb-11e6-9135-2f858a16f037.png">

**Comment by thejameskyle** [comment]
@sindresorhus did you intend to close this via #284?

**Comment by sindresorhus** [comment]
Nope. This is complementary to #284.

**Comment by hkdobrev** [comment]
@DrewML Do you still have plans to publish your work regarding the API token?

**Comment by bfred-it** [comment]
I see this as complementary to #181 into its own "Notifications Preview for GitHub" extension.
